### ISSUE TITLE
threathunting_file_summary_index is not populated

### ISSUE THREAD

**Issue by Mark-Law** [olafhartong/ThreatHunting#87]
I've searched through all of the XML and CONF files in the ThreatHunting application and cannot find how the summary index is being populated. Is there additional configuration to populate this index that is not in the base documentation? Because the index is not being populated, several dashboards are not returning information that I believe were intended to show additional information.

**Comment by hassj** [comment]
the same issue, 
i've already get many event from threathunting index, but have not anything index is populated on "Threat Hunting trigger Overview" dashboard.
Anybody kindly help

**Comment by dstaulcu** [comment]
The threathunting index is populated via collect commands which are part of scheduled searches defined in [savedsearches.conf](https://github.com/olafhartong/ThreatHunting/blob/master/default/savedsearches.conf).

As you browse through those searches note the use of macros such as `sysmon` , `windows-security` , `threathuting_index` and so on.  

Take a look at [macros.conf](https://github.com/olafhartong/ThreatHunting/blob/master/default/macros.conf) to see what those macros resolve to by default.   

Take a look at [readme.md](https://github.com/olafhartong/ThreatHunting/blob/master/README.md) for instructions to personalize the index, source, and sourcetypes as appropriate within your environment.

**Comment by hassj** [comment]
Thank for your reply @dstaulcu.
everything you said is fine, only "whitelist csv file Created/Installed" is missing although upload empty lookup folder to Splunk.
or i missing something else.

![Whitelist-missing](https://user-images.githubusercontent.com/47651127/193729098-1b7abd1c-bd67-4a45-bbf1-96e0dde20fbe.png)

**Comment by dstaulcu** [comment]
The output of savedsearch titled "[T1036] Masquerading - renamedbin" should be common when no whitelist entries are defined.  Cooy the search string associated with that and paste it into search input in threathunting namespace (app).  Remove last command in search redirecting output to threathunting index via collect command.  If you execute that over say last 24 hours do you get results?  If not, remove one command at a time and try again.  Do you eventually get results?

**Comment by hassj** [comment]
> The output of savedsearch titled "[T1036] Masquerading - renamedbin" should be common when no whitelist entries are defined. Copy the search string associated with that and paste it into search input in threathunting namespace (app). Remove last command in search redirecting output to threathunting index via collect command. If you execute what remains over all time do you get results? If not, remove one command at a time and try again. Do you eventually get results?

yes, it have result.

**Comment by hassj** [comment]
i also missing whitelist CVS on board as bellow: 
![image](https://user-images.githubusercontent.com/47651127/193738956-5262970b-fc19-4225-a892-9074ba64dadf.png)
show me some keypoint for resolving it @dstaulcu , 
thank in advance

**Comment by dstaulcu** [comment]
You can review the source of the [about](https://github.com/olafhartong/ThreatHunting/blob/master/default/data/ui/views/about.xml) dashboard to examine the underlying search query for the "Whitelist files created/installed.." panel throwing errors for you.

Run this query: "| rest /servicesNS/-/-/data/lookup-table-files"  -- Do you get results?
Run this query: "| rest /servicesNS/-/ThreatHunting/configs/conf-macros"  -- Do you get results?  I'm guessing you won't and that your macros.conf file has issues either relating to corruption or permissions.  
Run this query: "| rest /servicesNS/-/search/configs/conf-macros"  -- Do you get results?  I'm guessing you will.  If you do get results this affirms theory above that your ThreatHunting app install does not have integrity.   If you do not get results then that would suggest some underling access issue to the specific REST endpoint indicating integrity issues with your Splunk server itself.

**Comment by dstaulcu** [comment]
> > The output of savedsearch titled "[T1036] Masquerading - renamedbin" should be common when no whitelist entries are defined. Copy the search string associated with that and paste it into search input in threathunting namespace (app). Remove last command in search redirecting output to threathunting index via collect command. If you execute what remains over all time do you get results? If not, remove one command at a time and try again. Do you eventually get results?
> 
> yes, it have result.

Great -  if you are getting results now add the "| collect `threathunting_index`" command to the end of the search and run it again.  Do you get an error?   I'm guessing this relates back to some sort of problem with Splunk interacting with your macros.conf file as indicated by errors in Whitelist files created/installed error messages.

**Comment by hassj** [comment]
> i also missing whitelist CVS on board as bellow: ![image](https://user-images.githubusercontent.com/47651127/193738956-5262970b-fc19-4225-a892-9074ba64dadf.png) show me some keypoint for resolving it @dstaulcu , thank in advance

i've already fixed it by install ThreatHunting version 1.4.

**Comment by hassj** [comment]
> 



> > > The output of savedsearch titled "[T1036] Masquerading - renamedbin" should be common when no whitelist entries are defined. Copy the search string associated with that and paste it into search input in threathunting namespace (app). Remove last command in search redirecting output to threathunting index via collect command. If you execute what remains over all time do you get results? If not, remove one command at a time and try again. Do you eventually get results?
> > 
> > 
> > yes, it have result.
> 
> Great - if you are getting results now add the "| collect `threathunting_index`" command to the end of the search and run it again. Do you get an error? I'm guessing this relates back to some sort of problem with Splunk interacting with your macros.conf file as indicated by errors in Whitelist files created/installed error messages.

I got metric on "Activity by time per day" board but "Top triggered host_fqdns in the selected timeframe" and the rest boards still empty. 
![image](https://user-images.githubusercontent.com/47651127/194030350-602b8479-af33-472a-bc20-fc02eb57c8a3.png)

![image](https://user-images.githubusercontent.com/47651127/194031199-6562c6c3-46ea-4002-b6e7-c52aa2f6b163.png)
trying search with index=threathunting just get only one host that is splunk enterprise server, the other server which installed Universal Forwader have not.

**Comment by dstaulcu** [comment]
See discussion in issue #102. 

The GitHub version of threathunting app is far ahead of the splunkbase versions and corrects for many issues which I fear you will encounter next.

**Comment by hassj** [comment]
> 

Thank you for your reply. 
but everything's not going as well, I think search command on that boards have something wrong.
My topo: 
- one SE server (running on Ubuntu 20 included: ThreatHunting app and all dependence apps, sysmon ...) 
- one Client ( running windows 10 included Universal Forwarder)
after simulatting attack (running on local client) i see indicator on "Activity by time per day" board. that's it. and the rest board dont have anything as bellow: 
![image](https://user-images.githubusercontent.com/47651127/194472352-26579054-cab3-47fe-bcdb-9dcad4371827.png)

![image](https://user-images.githubusercontent.com/47651127/194472373-471f9c7d-f518-4918-bd65-1458207ac1c6.png)

Any help would be appreciated.

**Comment by dstaulcu** [comment]
I just installed the latest version of the app from GitHub (not Splunkbase) on a new search head.  
After creating the expected indexes and updating index names in macros as appropriate I started seeing events in the trigger overview dashboard as expected within an hour.   
There was one bug in the "required app status" panel of the "about this app" dashboard.  That bug was a problem in requirements.csv which I created two months ago.  I just submitted pull request #104 to correct for that and some other small things which should not affect your situation.   I don't think there is anything further I can/will do to help your situation.
